swap_meet/vendor.py

```diff
@@ -1,2 +1,84 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        category_items = []
+
+        for item in self.inventory:
+            if item.category == category:
+                category_items.append(item)
+
+        return category_items
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            other_vendor.inventory.remove(their_item)
+            other_vendor.inventory.append(my_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, other_vendor):
+        if self.inventory and other_vendor.inventory:
+            my_item = self.inventory[0]
+            their_item = other_vendor.inventory[0]
+            self.swap_items(other_vendor, my_item, their_item)
+            return True
+        else:
+            return False
+
+    def get_best_by_category(self, category):
+        max_condition = 0
+        best_item = None
+        for item in self.inventory:
+            if item.category == category:
```

Comment:
Consider calling your instance method `get_by_category()` so that you don't need to iterate through items in inventory and check if the category matches.

```python 
def get_best_by_category(self, category):
        max_condition = 0
        best_item = None
        items_by_category = self.get_by_category(category)
        for item in items_by_category:
            # the rest of your logic
```